app/helpers/react_on_rails_helper.rb

```diff
@@ -90,10 +90,18 @@ def define_render_if_dom_node_present(react_component, data_variable, dom_id, tr
     <<-JS.strip_heredoc
       var renderIfDomNodePresent = function() {
         var domNode = document.getElementById('#{dom_id}');
+
         if (domNode) {
           #{debug_js(react_component, data_variable, dom_id, trace)}
-          var reactComponent = #{react_component}(#{data_variable});
-          React.render(reactComponent, domNode);
+          var element;
+
+          if (Object.getPrototypeOf(#{react_component}) === React.Component) {
```

Comment:
I could confirm that it doesn't work with `createClass`. http://jsbin.com/relijopuqi/edit?js,console,output